### PR TITLE
absa model

### PR DESCRIPTION
**Pull request by MrPupik**

### CODE REVIEW

File: models/news_sentiment.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        text = text.lower()
+
+        score = infer(text, name)
+        text_score_df.append(pd.DataFrame({"title": [row["title"]], "date": [row['date']], "pos": [
+            score["pos"]], "neg": [score["neg"]], "neu": [score["neu"]]}))
+    return text_score_df
+
+
+def calc_compound_score(pos, neu, neg):
+    return tanh(2 * (pos - neg)*neu)
```
Comment (idobiran11) [inline models/news_sentiment.py]: Why multiply? didnt you want to divide?

File: models/news_sentiment.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    get a word and text splitted to sentences.
+    return list of sentences containing the word, each sentece score and text total score
+    """
+    relevant_corpus = []
+    scores = []
+    total_score = {'neg_s': 0.0, 'neu_s': 0.0, 'pos_s': 0.0, 'compound_s': 0.0}
+    text_sent = sent_tokenize(text)
+    num_of_sentences = len(text_sent)
+    for text in text_sent:
+        name_pos = text.find(name)
```
Comment (idobiran11) [inline models/news_sentiment.py:134]: Does this work on capital and lowercase?

File: models/news_sentiment.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+# df = pd.read_csv("data/fox-articles-netanyahu.csv")
+
+
+def news_sentiment_handler(object_name, news_vendor, corpus, output_directory="data/output_data"):
+    def lda(df: pd.DataFrame):
+
+        df.text = df.text.apply(lambda text: text.replace(
+            '\n', ' ').replace('\xa0', ' ').replace('  ', ' '))
+        df = df.rename(columns={'timestamp': 'date'})
```
Comment (idobiran11) [inline models/news_sentiment.py]: What does this do?

File: models/simple_model_notebook.py [hunk 4]
```diff
@@ -22,73 +23,23 @@
...
-    corpus.rename(columns={'timestamp': 'date'}, inplace=True)
-    scores = calc_scores_on_corpus(corpus, object_name)
-    scores_graph = plot_graphs(scores, object_name, news_vendor)
-    scores_graph.set_index('index')
-    print_max_min_articles(scores_graph, corpus)
-    filepath = f"{output_directory}/{news_vendor}_{object_name}_nltk_sentences_sentiment.csv"
-    scores_graph.to_csv(filepath, index=False)
-    neptune_run[f'eval/{news_vendor}_sentiment'].upload(filepath)
-    return scores_graph
+    def lda(df): return df.rename(columns={'timestamp': 'date'})
```
Comment (idobiran11) [inline models/simple_model_notebook.py]: Not sure i understand what's happening here

File: models/news_sentiment.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            relevant_corpus.append(text)
+            curr_score = infer(text, name)
+            scores.append(curr_score)
+
+            total_score["neg_s"] += curr_score["neg"]
+            total_score["neu_s"] += curr_score["neu"]
+            total_score["pos_s"] += curr_score["pos"]
+
+            total_score["compound_s"] += calc_compound_score(
+                curr_score["pos"], curr_score["neu"], curr_score["neg"])
```
Comment (idobiran11) [inline models/news_sentiment.py:146]: So here for each sentence with the word, you add the value to the total score, and then calculate the compound score?

File: models/news_sentiment.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+    total_score["neg_s"] = total_score["neg_s"] / num_of_sentences
+    total_score["neu_s"] = total_score["neu_s"] / num_of_sentences
+    total_score["pos_s"] = total_score["pos_s"] / num_of_sentences
+    total_score["compound_s"] = total_score["compound_s"] / num_of_sentences
```
Comment (idobiran11) [inline models/news_sentiment.py]: And this is what you spoke with elad in the group right? It really makes more sense to divide only by the sentences where the word exists doesn't it?

File: models/news_sentiment.py (hunk 2 above)
Comment (MrPupik) [reply models/news_sentiment.py:134]: text and name get `lower` before:
image

File: models/news_sentiment.py (hunk 3 above)
Comment (MrPupik) [reply models/news_sentiment.py]: news sentiment model get's fucked up when these chars show up (eda)

File: models/news_sentiment.py (hunk 5 above)
Comment (MrPupik) [reply models/news_sentiment.py:146]: yes but i calc `compund` from `curr_score`, not from `total`

File: models/news_sentiment.py (hunk 6 above)
Comment (MrPupik) [reply models/news_sentiment.py]: fixed it

File: preprocess/coreference_resolution.py [hunk 7]
```diff
@@ -15,22 +15,28 @@ def no_preprocess(news_vendor: str, object_name: str, input_csv_name: str,
...
                      output_dir: str = "data/preprocessed_data"):
-    nlp = spacy.load("en_coreference_web_trf")
+
     df = pd.read_csv(f'{DATA_DIR}/{input_csv_name}')
     output_filename = f'{news_vendor}_{object_name}_{PreprocessNames.COREF}.csv'
     full_output_path = f'{output_dir}/{output_filename}'
-    if not os.path.isfile(full_output_path):
+
+    if os.path.isfile(full_output_path):
+        df = pd.read_csv(full_output_path)
```
Comment (MrPupik) [inline preprocess/coreference_resolution.py:24]: @idobiran11 
you see what i did here? it's ok right?

Comment (idobiran11) [reply preprocess/coreference_resolution.py:24]: ya good

**Pull request merged**